engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/util/GraphQLUtil.kt

```diff
@@ -164,10 +164,8 @@ internal fun mergeResults(results: List<ExecutionResult>): ExecutionResult {
     val errors: MutableList<GraphQLError> = mutableListOf()
 
     for (result in results) {
-        when (val resultData = result.getData<Any?>()) {
-            null -> {
-            }
-            is AnyMap -> data.hackPutAll(resultData)
+        when (val resultData = result.getData<JsonMap?>()) {
+            is AnyMap -> updateOverallResultAndMergeSameNameTopLevelFields(data, resultData)
```

Comment:
I understand this came about because of refactoring, but this logic reads really really weirdly. Effectively you're just trying to check if `resultData` is NOT null.

I would just do

```
val resultData = result.getData<JsonMap?>()
if (resultData != null) {
  updateOverallResultAndMergeSameNameTopLevelFields(data, resultData)
}
```